language_detector/main.py

```diff
@@ -1,9 +1,58 @@
+# -*- coding: utf-8 -*-
 """This is the entry point of the program."""
 
 from .languages import LANGUAGES
-
+import string
+import re
 
 def detect_language(text, languages):
-    """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    #solution 2
+    '''
+    regex = re.compile('[%s]' % re.escape(string.punctuation))
+    ls = regex.sub('',text).split()
+    word_counts = {}
+    freq = {}
+    for word in ls:
+        word_counts[word] = word_counts.get(word,0) + 1
+    for dict in languages:
+        count = 0
+        for key,value in word_counts.items():
+            if key in dict["common_words"]:
+                count += value
+                freq[dict["name"]] = count
+    return max(freq, key=freq.get)
+    '''
+    
+    #solution 1
+    #Set Dictionary and Variable
+    stats = {}
+    found = 0
+    regex = re.compile('[%s]' % re.escape(string.punctuation))
+    text = regex.sub('',text)
+
+    #For each word found in split of the text variable passed
+    for word in text.split():
+
+        #For each item in the length of languages list
+        for i in range(len(languages)):
+
+            #For each word in the common_words list for the current language
+            #as determined by i from the prior for statement.
+            for lang in languages[i]["common_words"]:
+
+                #If the word from the text list == the current lang word from
+                #the common_words list for the current language as determined
+                #by i from the prior for statement.
+                if word.lower() == lang.lower():
+
+                    #Add 1 to found and update stats dictionary with matching
+                    #key of the current language
+                    found += 1
```

Comment:
``found`` is accumulating points for all languages as the program iterates over the the words in ``text``.  This could yield  a bug.  Think about how the program would score each of these sentences:

*Ayer tu y yo fuimos a Madrid.*

*Yesterday your mother and I went to see a play.*